consensus/state_processing/src/per_block_processing/eip4844/eip4844.rs

```diff
@@ -0,0 +1,119 @@
+use crate::BlockProcessingError;
+use eth2_hashing::hash_fixed;
+use itertools::{EitherOrBoth, Itertools};
+use ssz::Decode;
+use ssz_types::VariableList;
+use types::consts::eip4844::{BLOB_TX_TYPE, VERSIONED_HASH_VERSION_KZG};
+use types::{
+    AbstractExecPayload, BeaconBlockBodyRef, EthSpec, ExecPayload, FullPayload, FullPayloadRef,
+    KzgCommitment, Transaction, Transactions, VersionedHash,
+};
+
+pub fn process_blob_kzg_commitments<T: EthSpec, Payload: AbstractExecPayload<T>>(
+    block_body: BeaconBlockBodyRef<T, Payload>,
+) -> Result<(), BlockProcessingError> {
+    if let (Ok(payload), Ok(kzg_commitments)) = (
```

Comment:
We can probably `?` these away right? Once 4844 is activated we can expect both `payload` + `blob_kzg_commitments` can't we?

Reply:
There's no check elsewhere in block processing before this that we're in 4844 so if we error here we couldn't process pre 4844 blocks - but this raises the question of how we should be handling fork specific logic in the `per_block_processing` method generally cause it isn't obvious that it's shared and what logic exactly is shared across what fork. Maybe the current set up of "process this field if it exists" is the best solution but I'm not sure